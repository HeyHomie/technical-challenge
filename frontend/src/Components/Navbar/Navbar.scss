.navbar {
  align-items: center;
  background-color: var(--navbar);
  color: var(--primary);
  display: flex;
  font-size: 14px;
  justify-content: space-between;
  line-height: 1.5;
  padding: 12px 32px;

  &-responsive__container {
    @media screen and (min-width: 768px) {
      display: none !important;
    }
  }

  &-container {
    align-items: center;
    display: flex;

    &__responsive-icon {
      fill: var(--white);

      summary {
        font-size: 0;
      }

      @media screen and (min-width: 768px) {
        display: none;
      }
    }

    &__logo {
      a {
        fill: var(--white);
        margin-right: 16px;
      }
    }

    &__notifications,
    &__summary {
      margin-left: 16px;
      fill: var(--white);
      @media screen and (max-width: 767px) {
        margin: 0;
      }
    }

    &__summary {
      @media screen and (max-width:767px) {
        display: none;
      }
    }

    &__notifications {
      position: relative
    }

    &__notifications-active {
      background-color: var(--blue);
      border-radius: 50%;
      height: 8px;
      position: absolute;
      right: 0;
      top: -5px;
      width: 8px;
    }

    &__searchbar {
      form {
        align-items: center;
        display: flex;
        position: relative;

        input {
          background-color: transparent;
          border: 1px solid #57606a;
          border-radius: 6px;
          font-size: inherit;
          min-height: 26px;
          outline: none;
          padding: 5px 12px;
          width: -webkit-fill-available;
          &::-webkit-input-placeholder {
            color: #c2c2c2e5;
          }
        }


        svg {
          position: absolute;
          right: 4px;
        }
      }
      @media screen and (max-width: 767px) {
        display: none;
      }
      @media screen and (min-width: 768px) {
        max-width: 272px;
        overflow: hidden;
        width: 272px
      }
    }
    &__listItem {
      align-items: center;
      display: flex;
      margin: 0;
      padding: 0;

      li {
        list-style: none;

        a {
          color: var(--white);
          font-size: 14px;
          font-weight: 600;
          text-decoration: none;
          line-height: 1.5;
          margin-left: 16px;
        }
      }
      @media screen and (max-width: 767px) {
        display: none;
      }
    }
  }
}
