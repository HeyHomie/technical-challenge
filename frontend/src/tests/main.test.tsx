import React from 'react'
import { Router } from 'react-router-dom'
import { createMemoryHistory } from 'history'
import renderer from 'react-test-renderer'

import Navbar from '../components/layout/navbar'

const history = createMemoryHistory()

test('Renders correctly', () => {
  const component = renderer.create(
    <Router history={history}>
      <Navbar />
    </Router>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  expect(tree).toBeTruthy()
})
