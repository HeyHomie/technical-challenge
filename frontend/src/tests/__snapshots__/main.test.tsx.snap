// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<nav
  className="w-full shadow-sm  py-3 px-1.5 h-16 bg-gray-800 text-white"
>
  <div
    className="flex max-w-screen-2xl m-auto mt-2 justify-between items-center"
  >
    <div
      className="flex items-center px-6"
    >
      <div
        className="pr-6"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
    </div>
  </div>
</nav>
`;
